src/wifi/model/mac-low.cc

```diff
@@ -2177,58 +2177,44 @@ Time
 MacLow::CalculateDmgTransactionDuration (Time packetDuration)
 {
   NS_LOG_FUNCTION (this << packetDuration);
-  Time duration = packetDuration;
-  if (m_txParams.MustWaitNormalAck ())
-    {
-      duration += GetAckDuration (m_currentHdr.GetAddr1 (), m_currentTxVector);
-    }
-  else if (m_txParams.MustWaitFastAck () || m_txParams.MustWaitSuperFastAck ())
-    {
-      duration += GetPifs ();
-    }
-  if (m_txParams.MustWaitBasicBlockAck ())
-    {
-      duration += GetSifs ();
-      WifiTxVector blockAckReqTxVector = GetBlockAckTxVector (m_currentHdr.GetAddr2 (), m_currentTxVector.GetMode ());
-      duration += GetBlockAckDuration (blockAckReqTxVector, BASIC_BLOCK_ACK);
-    }
-  else if (m_txParams.MustWaitCompressedBlockAck ())
-    {
-      duration += GetSifs ();
-      WifiTxVector blockAckReqTxVector = GetBlockAckTxVector (m_currentHdr.GetAddr2 (), m_currentTxVector.GetMode ());
-      duration += GetBlockAckDuration (blockAckReqTxVector, COMPRESSED_BLOCK_ACK);
-    }
-  return duration;
+  return CalculateDmgTransactionDuration (packetDuration, m_currentHdr, m_txParams);
 }
 
 Time
-MacLow::CalculateDmgTransactionDuration (Ptr<Packet> packet, WifiMacHeader &hdr)
+MacLow::CalculateDmgTransactionDuration (Ptr<Packet> packet, WifiMacHeader &hdr) 
 {
   NS_LOG_FUNCTION (this << packet << hdr);
   Time duration = m_phy->CalculateTxDuration (GetSize (packet, &hdr, m_ampdu),
                                               m_currentTxVector, m_phy->GetFrequency ());
-  if (m_txParams.MustWaitNormalAck ())
+  return CalculateDmgTransactionDuration (duration, hdr, m_txParams);
+}
+
+Time
+MacLow::CalculateDmgTransactionDuration (Time packetDuration, WifiMacHeader &hdr, MacLowTransmissionParameters &txParams) const
```

Comment:
Did you create this function with a different signature because you needed another method with a same name? Otherwise I cannot figure out why you would need to pass as parameter a member variable, when you can directly use it. I think it might be useful only if you need to modify a copy of the member variable, otherwise I would try to find a more clever way to do it. We can discuss it offline

Reply:
he uses it in MacLow::StopMpduAggregation, i was also about to ask this before i checked the other code :)